src/main/java/uk/gov/hmcts/reform/civil/HighLevelDataSetupApp.java

```diff
@@ -82,7 +83,12 @@ public void createRoleAssignments() {
     }
 
     @Override
-    protected boolean shouldTolerateDataSetupFailure() {
-        return true;
+    protected boolean shouldTolerateDataSetupFailure(Throwable e) {
+        int httpStatusCode504 = 504;
+        if (e instanceof ImportException) {
+            ImportException importException = (ImportException) e;
+            return importException.getHttpStatusCode() == httpStatusCode504;
```

Comment:
Instead of declaring an extra variable, I would just put: 
return importException.getHttpStatusCode() == 504;